File: ios/CredentialSharing.m
#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(CredentialSharing, NSObject)

RCT_EXTERN_METHOD(getSharedCredentials:(String)a withB:(String)b withC:(String)c
                  withResolver:(RCTPromiseResolveBlock)resolve
                  withRejecter:(RCTPromiseRejectBlock)reject)

@end

File: ios/CredentialSharing.swift
import Foundation

@objc(CredentialSharing)
class CredentialSharing: NSObject {

    @objc(getSharedCredentials:withB:withC:withResolver:withRejecter:)
    func getSharedCredentials(
     service: String, account: String, accessGroup: String,
     resolve: RCTPromiseResolveBlock,
     reject: RCTPromiseRejectBlock) -> Void {
         resolve(CredentialSharingKeychainAccess.shared.accessToken(service: service, account: account, accessGroup: accessGroup))
    }
}
